src/tree/render.js

```diff
@@ -27,84 +26,61 @@ const filterTree = (id, nodeTree) => {
   });
 };
 
-export const reRenderTree = ({
-  svg,
-  divBox,
-  activeNode,
-  allNodes,
-  o,
-  width,
-  height,
-  cardStyling,
-  callback,
-  storage,
-}) => {
-  const nodes = filterTree(activeNode, allNodes);
+export const paintTree = ({ objectData, activeNode, styling, setActiveCallback }) => {
+  const { svg, divBox, allNodes, positioning, width, height } = objectData;
   divBox.selectAll('.node-rect').remove();
   svg.selectAll('g').remove();
-
-  // Create the nodes.
+  const nodes = filterTree(activeNode, allNodes);
+  // create the nodes
   const node = svg
     .selectAll('.node')
     .data(nodes)
     .enter()
     .append('g')
     .attr('class', 'nodeWrapper')
     .attr('id', d => d.data.id);
-
-  box(divBox, o, nodes, cardStyling, id => {
-    callback({
-      svg,
-      divBox,
-      allNodes,
-      o,
-      activeNode: id,
-      width,
-      height,
-      cardStyling,
-      callback,
-      storage,
-    });
+  // Create cards
+  box(divBox, positioning, nodes, styling, id => {
+    setActiveCallback(id);
```

Comment:
I guess this behaviour will change a bit with the changes from Elise, but this looks already nicer I would say

Reply:
Yes. We should send stuff back to index here and then run an update from there.